nix/test_devshell.py

```diff
@@ -0,0 +1,81 @@
+import os
+import shutil
+import subprocess
+import pytest
+
+
+def translate_lc(lc: str):
+    # awslc for S2N_LIBCRYPTO is special.
+    if "awslc" in lc:
+        return "aws-lc"
+    else:
+        return lc
+
+
+@pytest.mark.parametrize("lc", [os.getenv("S2N_LIBCRYPTO")])
+def test_s2n_libcrypto(lc):
+    # Validate S2N_LIBCRYPTO is in the CMAKE_INCLUDE_PATH
+    libcrypto = translate_lc(lc)
+    include_path = os.getenv("CMAKE_INCLUDE_PATH")
+    assert libcrypto is not None
+    assert include_path is not None
+    assert libcrypto in include_path
+
+
+@pytest.mark.parametrize("lc", [os.getenv("S2N_LIBCRYPTO")])
+def test_s2n_libcrypto_uniq(lc):
+    # Make certain we only have the preferred libcrypto in CMAKE_INCLUDE_PATH.
+    libcrypto = translate_lc(lc)
+    include_path = os.getenv("CMAKE_INCLUDE_PATH")
+    assert libcrypto is not None
+    assert include_path is not None
+    all_lcs = ["openssl-3.0", "openssl-1.1.1",
+               "openssl-1.0.2", "libressl", "aws-lc"]
```

Comment:
Should we also maybe have a test that asserts that S2N_LIBCRYPTO is in this list? That way when we add a new libcrypto the test will break and we will know to update this list for both tests. Maybe this isn't needed, though, since we don't add libcryptos very often.

Reply:
Since there are tests where S2N_LIBCRYPTO is used, I'm all for checking to make sure it's set to something expected.